keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1079 @@
+      - serviceAccountToken:
+          path: wi-for-image-pull
+          audience: my-audience # must match the credential provider audience
+```
+
+Example credential provider configuration:
+
+```yaml
+apiVersion: kubelet.config.k8s.io/v1
+kind: CredentialProviderConfig
+providers:
+  - name: acr-credential-provider
+    matchImages:
+      - "*.registry.io/*"
+    defaultCacheDuration: "10m"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1
+    audience: my-audience
+```
+
+#### Credential Provider Request API
+
+Add `ServiceAccountToken` and `PodAnnotations` fields to `CredentialProviderRequest`.
+
+1. `ServiceAccountToken` is the service account token bound to the pod for which the image is being pulled.
+    If the `Audience` field is configured in the kubelet's credential provider configuration, the token will be sent to the plugin.
+2. `PodAnnotations` is a map of annotations on the pod for which the image is being pulled.
+    This field is optional and may be empty. Plugin may use this field to extract additional information required to fetch credentials.
+    For example, a plugin may use annotations to determine the identity ID to use when fetching credentials.
+
+<!-- Would it be possible for a plugin to talk to kubelet API and get the info instead of passing the pod annotations? -->
+<!-- We already send pod annotations to CRI, so this is consistent with that -->
+
+<!-- Add note about a plugin that just returns the service account token without any additional logic and that gets passed to the CRI. -->
+
+```diff
+--- a/staging/src/k8s.io/kubelet/pkg/apis/credentialprovider/types.go
++++ b/staging/src/k8s.io/kubelet/pkg/apis/credentialprovider/types.go
+@@ -32,6 +32,16 @@ type CredentialProviderRequest struct {
+        // credential provider plugin request. Plugins may optionally parse the image
+        // to extract any information required to fetch credentials.
+        Image string
++
++       // serviceAccountToken is the service account token bound to the pod for which
++       // the image is being pulled. This token is only sent to the plugin if the
++       // audience field is configured in the kubelet's credential provider configuration.
++       ServiceAccountToken string
++
++       // podAnnotations is a map of annotations on the pod for which the image is being pulled.
++       // This field is optional and may be empty. Plugins may use this field to extract
++       // additional information required to fetch credentials.
++       PodAnnotations map[string]string
+ }
+```
+
+#### Credential Provider Response API
+
+When the `audience` field is set in the kubelet's credential provider configuration, the plugin must return a `CredentialProviderResponse` with the `cacheMode` field set to `ServiceAccount` or `None`. If the cache mode is not set, kubelet will error out and not use the response. The field is required and serves as a marker to indicate that the plugin understands the new fields in the request.
```

Comment:
I think it's questionable to send service account information (either dimensions or an actual token) to a credential provider and make it possible for it to return a result that is cached across service accounts